api/operator_autopilot_test.go

```diff
@@ -181,4 +181,9 @@ func TestAPI_OperatorAutopilotServerHealth_429(t *testing.T) {
 	out, err := client.Operator().AutopilotServerHealth(nil)
 	require.NoError(t, err)
 	require.Equal(t, &reply, out)
+
+	mapi.withReply("GET", "/v1/operator/autopilot/health", nil, 500, nil).Once()
+	out, err = client.Operator().AutopilotServerHealth(nil)
+
+	require.Equal(t, 500, err.(StatusError).Code)
```

Comment:
not much of a fan of the UX here -- the original proposal on https://discuss.hashicorp.com/t/golang-consul-api-pulling-out-the-statuscode-from-deeply-nested-responses/18037 was a more straightforward approach than the type assertions that devs would need to do here.

what does everyone think about this?

Reply:
I would find it surprising to include the HTTP status code as part of the response type.

Here's how I would expect someone to use the `StatusError`:

```go
var status *statusError
if errors.As(err, status) {
    // do something with status.Code
}
```

The test should probably do something similar with `errors.As` or `errors.Is` to prevent a panic when the error is of a different type.

Reply:
thanks for pointing that out !! 💯 

with that more idiomatic way of inspecting the err type, I think I'm less concerned about the UX here!